lib/plugins/aws/deployFunction.js

```diff
@@ -103,10 +120,13 @@ class AwsDeployFunction {
       FunctionName: functionObj.name,
     };
 
-    if ('awsKmsKeyArn' in functionObj && !_.isObject(functionObj.awsKmsKeyArn)) {
-      params.KMSKeyArn = functionObj.awsKmsKeyArn;
-    } else if (serviceObj && 'awsKmsKeyArn' in serviceObj && !_.isObject(serviceObj.awsKmsKeyArn)) {
-      params.KMSKeyArn = serviceObj.awsKmsKeyArn;
+    let kmsKeyArn = functionObj.kmsKeyArn || providerObj.kmsKeyArn;
```

Comment:
I think this could be simplified even further with `kmsKeyArn = functionObj.kmsKeyArn || providerObjkmsKeyArn || functionObj.awsKmsKeyArn || serviceObj.awsKmsKeyArn;